### ISSUE TITLE
Some streams can return 404

### ISSUE THREAD

**Issue by Tyrrrz** [Tyrrrz/YoutubeExplode#759]
### Version

master

### Platform

.NET 8

### Steps to reproduce

```csharp
var streamManifest = await _youtube.Videos.Streams.GetManifestAsync("JQgKhZZyBYg");
var streamInfo = streamManifest.GetMuxedStreams().MaxBy(s => s.VideoQuality);

// This fails
await _youtube.Videos.Streams.DownloadAsync(streamInfo, "video.mp4");
```

### Details

Looks like one of the muxed streams (720p mp4) returns 404 when accessed.

Originally reported at https://github.com/Tyrrrz/YoutubeDownloader/issues/406

### Checklist

- [X] I have looked through existing issues to make sure that this bug has not been reported before
- [X] I have provided a descriptive title for this issue
- [X] I have made sure that this bug is reproducible on the latest version of the package
- [X] I have provided all the information needed to reproduce this bug as efficiently as possible
- [X] I have sponsored this project

**Comment by Tyrrrz** [comment]
A bit more detail into what's happening. The stream _does not_ immediately return 404, which is why it's not filtered out by this check in the `StreamClient`:

https://github.com/Tyrrrz/YoutubeExplode/blob/49ce82843504ffe0f76d476620a9cef68b83275c/YoutubeExplode/Videos/Streams/StreamClient.cs#L85-L87

Instead, what actually happens is that the stream initially serves data, up to a certain point:

![image](https://github.com/Tyrrrz/YoutubeExplode/assets/1935960/c22bfd7b-6673-4f1d-9e47-d4d352cf6de6)
![image](https://github.com/Tyrrrz/YoutubeExplode/assets/1935960/e6b429c8-2b7d-4a6d-80be-f39ff3550793)

Then, at some point, the underlying `ContentLengthReadStream` class throws an exception, indicating that the stream has mismatched content length:

![image](https://github.com/Tyrrrz/YoutubeExplode/assets/1935960/e0c2a58b-9109-4769-a661-0f19d3bc4d4f)

After that, `MediaStream` attempts to re-initialize the last segment, on the off chance that this is a transient error. This time, the stream returns 404 instead, most likely because the new values of the `range` parameter are out of bounds.

Worth noting that even Chrome fails to read this stream, due to content length mismatch:

![image](https://github.com/Tyrrrz/YoutubeExplode/assets/1935960/2119073a-67ac-495b-9a34-eed2c9a47090)

The bottom line is that this stream is broken, but it's currently not filtered out because the issue appears later on in the stream's lifecycle.

**Comment by QINGCHARLES** [comment]
Wow, what a weird error. I can only guess YouTube's encoder failed for some reason and pushed out a broken file that none of their systems are set up to check for.

There is not much for you to do in your code. The best you could do is indicate to the user somehow that they should try the stream again but select a different stream type to download and it might work.

**Comment by Tyrrrz** [comment]
Looks like I found a way to filter them out. I'll release a new update on Thursday.

**Comment by QINGCHARLES** [comment]
You're a star, thank you.
